src/main/java/uk/gov/hmcts/darts/common/service/impl/JudgeCommonServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.darts.common.service.impl;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.darts.common.entity.JudgeEntity;
+import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
+import uk.gov.hmcts.darts.common.repository.JudgeRepository;
+import uk.gov.hmcts.darts.common.service.JudgeCommonService;
+
+import java.util.Locale;
+import java.util.Optional;
+
+@RequiredArgsConstructor
+@Service
+public class JudgeCommonServiceImpl implements JudgeCommonService {
+
+    private final JudgeRepository judgeRepository;
+
+    @Override
+    @Transactional
+    public JudgeEntity retrieveOrCreateJudge(String judgeName, UserAccountEntity userAccount) {
+        if (judgeName == null || judgeName.trim().isEmpty()) {
```

Comment:
For future reference,this could have also been achieved using apache commons StringUtils.isBlank